apps/modernization-api/src/main/java/gov/cdc/nbs/config/KafkaConfig.java

```diff
@@ -37,38 +44,95 @@ public class KafkaConfig {
     @Value("${kafka.enabled:true}")
     private boolean kafkaEnabled;
 
-    @Bean
-    public NewTopic createPatientSearchTopic() {
-        if (!kafkaEnabled) {
-            return null;
-        }
-        return TopicBuilder.name(patientSearchTopic).partitions(topicPartitionCount).replicas(topicReplicationFactor)
-                .compact().build();
-    }
+	@Bean
+	public NewTopic createPatientSearchTopic() {
+		if (!kafkaEnabled) {
```

Comment:
The  `@ConditionalOnProperty`  annotation could be used at the class level to enable/disable the entire configuration.  It would be something like `@ConditionalOnProperty(prefix = "kafka", name = "enabled", havingValue = "true")`.  It would include the Kafka Templates also so if you go that route you might want to separate the creation of the topics from the templates as two Configurations.

Reply:
Hi @adamloup-enquizit  for now  I would prefer to keep pulling kafkaEnabled flag from the application.yaml setting and  after these changes merge I can look into using the @condittionalOnProperty flag separating the creation of the topics  from the templates  in to a separate PR would those updates tested separately.